### PR TITLE
Generate Nth Prime Number

### PR DESCRIPTION
**Pull request by Hawk453**
An algorithm to generate Nth Prime Number based on Sieve of Eratosthenes.

### CODE REVIEW

**Review by siriak — APPROVED** [review]

**Review by siriak — APPROVED** [review]
LGTM, could you fix the failing check, probably something's up because of merging in master

File: src/math/nthprime.rs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        total_prime = primes[2..].iter().sum();
+        size_factor += 1;
+        s = n * size_factor;
+    }
+
+    count_prime(primes, n).unwrap()
+}
+
+fn get_primes(s: u64) -> Vec<u64> {
+    let mut v: Vec<u64> = vec![1; s as usize];
```
Comment (er888kh) [inline src/math/nthprime.rs:24]: Can you change this to `Vec<bool>`?
It will help with memory usage (and maybe vectorization).

Comment (Hawk453) [reply src/math/nthprime.rs:24]: Certainly, will do this.

File: src/math/nthprime.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            }
+        }
+    }
+    v
+}
+
+fn count_prime(primes: Vec<u64>, n: u64) -> Option<u64> {
+    let mut counter: u64 = 0;
+    for i in 2..primes.len() {
+        counter += primes.get(i).unwrap();
```
[the remark this answers is not in the document]
Comment (Hawk453) [reply src/math/nthprime.rs:43]: clippy recommended get(i) but lets see, I'll do this too.

File: src/math/nthprime.rs (hunk 1 above)
Comment (Hawk453) [reply src/math/nthprime.rs:24]: Utilizing the vec in the function get_nth_prime. The while loop has a use case for the sum of the count.

File: src/math/nthprime.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hawk453) [reply src/math/nthprime.rs:43]: ```the loop variable `i` is used to index `primes` ``` Clippy warning when using primes[i]

**Comment by Hawk453** [comment]
@er888kh Looks like its fine as it is for both suggestions are redundant.

**Comment by siriak** [comment]
The check build / cargo fmt (pull_request)  is failing and the branch is out of date a bit

**Comment by er888kh** [comment]
> @er888kh Looks like its fine as it is for both suggestions are redundant.

Oh I see. So then just do what siriak suggested and call it a day :)

**Review by siriak — APPROVED** [review]
Awesome, thanks!

**Pull request merged**